### DISCUSSION TITLE
Compatibility problem.

### DISCUSSION THREAD

**Discussion by Mishasama** [mono/mono#21706] [Q&A]
I have read the documentation on the official website, which says that the highest compatibility is currently up to .NET 4.7.
I would like to ask why the Mono compatibility version is so far behind, given that .NET 6 is already an LTS version and .NET 7 is already a release version. Is there a plan to upgrade to a higher version?
Because the new versions of .NET have many performance and design optimizations, it would be a shame to give up the performance under Windows to be compatible with Linux.

I am sorry, I am just a beginner, and I am just getting to know Mono today. I am sorry if my question seems stupid.

**Comment by jbatt33** [comment]
Why not use dotnet runtime instead for .NET 6?

**Reply by Mishasama** [reply]
What do you mean?

**Reply by jbatt33** [reply]
.NET 6/7/8 are not .NET Framework, they are a melding of .NET core and .NET framework in an attempt to unify all of .NET. So if you want to use the newer language .NET (6/7/8) you should pick up one of the .NET runtimes. Microsoft provides a bunch of precompiled versions for various targets.

**Reply by Mishasama** [reply]
I… don't know. Maybe you can help us to make [this project](https://github.com/SubtitleEdit/subtitleedit/issues/7327#:~:text=4.0.0/SubtitleEditBeta.zip-,If%20I%20upgrade%20to%20.NET%207%20then%20Linux%20users%20will%20not%20be%20able%20to%20run%20SE...,-I%27ve%20tested%20upgrading) to the newer .NET version? Or teach us how?

**Reply by jbatt33** [reply]
Subtitle Edit would need to be migrated to dotnet. It is fine that they are using mono, there is likely little performance gain to move over. If anything I found mono uses less memory over dotnet.

**Reply by jbatt33** [reply]
Here is some further information to help you understand. 
https://learn.microsoft.com/en-us/dotnet/standard/choosing-core-framework-server
You'll see there that .NET Framework stopped at 4.8, while .NET (core) continued on.

**Reply by Mishasama** [reply]
I see that core seems to be better suited for cross-platform? I'm still a bit puzzled because it's very slow in processing files with thousands of lines over 1 hour. And it didn't use up my CPU performance.
But I've seen other software that handles large amounts of data upgraded from 4 all the way to 7 for performance. I'm not sure if it's worth it for SE, what do you think?
He also mentioned the[ WinForms issue](https://github.com/SubtitleEdit/subtitleedit/issues/7327#:~:text=Winforms%20dot%20net%206%2B%20is%20not%20ported%20to%20Linux%20as%20far%20as%20I%20know.), I wonder if that would be a hindrance to permanently not being able to upgrade?

**Reply by jbatt33** [reply]
If you are referring to the ability to read a file via parallel processing, that is available in .NET framework and is up to the developer of that project to provide this functionality. 

If you are concerned about OCR issues, I'm not sure there is anything to be done there.

**Reply by jbatt33** [reply]
@Mishasama  I think it would be worth seeing if the owner of that project wants to move over to .NET 8 and drop using mono as their cross platform runtime.

**Reply by Mishasama** [reply]
Is there any easy way to easily migrate a project to a newer version like .NET 8? I'm worried that he won't be able to do a huge migration job by himself.

**Reply by jbatt33** [reply]
There is a migration tool (I think) that will indicate all what is not compatible. I believe that tool is expected to be used to generate a task list of things to work on. It really depends on the project's use of .NET framework. You can clone the repo and see what happens when you target .NET 8. You'll need to create/modify project files to start :) 

Checking this page: https://learn.microsoft.com/en-us/dotnet/core/porting/

I'm wondering if the OCR plugin could be an issue: 
"Third-party controls and libraries might not have been ported to .NET and remain only available to .NET Framework." 
Unless the OCR plugin is native... 

I think we should move this discussion off the mono project and over to subtitle edit's repo. This really has nothing to do with mono.
